### PR TITLE
Take BFACTORY BLOCK into account.

### LINKED ISSUE #757: Update Read.me to reflect the encoding Aquarius accepts
Aquarius(&Provider) works with UTF8 encoding, update the Read.me to clearly reflect this

### PR DESCRIPTION
**Pull request by calina-c**
Closes #757.

### CODE REVIEW

File: aquarius/events/util.py [hunk 1]
```diff
@@ -47,6 +47,28 @@ def get_network_name():
...
+        if network_values["chainId"] == chain_id
+    )
+
+
+def get_defined_block(chain_id: int):
+    """Retrieves the block either from envvar, either from address.json file."""
+    with open(get_address_file(), "r") as address_file:
+        addresses = json.load(address_file)
+    return (
+        int(os.getenv("BFACTORY_BLOCK"))
```
Comment (calina-c) [inline aquarius/events/util.py:66]: if this exists, do not load/open the address file, there is no point.

File: aquarius/events/util.py [hunk 2]
```diff
@@ -47,6 +47,28 @@ def get_network_name():
...
+        if "BFACTORY_BLOCK" in os.environ
+        else addresses[get_network_name_by_chain_id(chain_id)]["startBlock"]
```
Comment (calina-c) [inline aquarius/events/util.py:68]: you don't need the network name. You can retrieve the start block by chain_id directly.

File: aquarius/events/events_monitor.py [hunk 3]
```diff
@@ -282,10 +285,15 @@ def get_last_processed_block(self):
             block = (
                 last_block_record["last_block"]
                 if last_block_record["last_block"] >= 0
-                else 0
+                else get_defined_block(self._chain_id)
             )
         except Exception as e:
-            logging.error(f"Cannot get last_block error={e}")
+            # Retrieve the defined block.
+            if type(e) == elasticsearch.NotFoundError:
```
Comment (calina-c) [inline aquarius/events/events_monitor.py:292]: not sure this is correct.

Comment (mariacarmina) [reply aquarius/events/events_monitor.py:292]: Why? if the error is `NotFoundError` from ES, should retrieve the `defined_ block`

File: aquarius/events/util.py [hunk 4]
```diff
@@ -47,6 +47,25 @@ def get_network_name():
...
+        for network_values in addresses.values()
+        if network_values["chainId"] == chain_id
+    )
+
+
+def get_defined_block(chain_id: int):
+    """Retrieves the block either from envvar, either from address.json file."""
+    if "BFACTORY_BLOCK" in os.environ:
+        return int(os.getenv("BFACTORY_BLOCK"))
+    else:
```
[the remark this answers is not in the document]
Comment (mariacarmina) [reply aquarius/events/util.py:65]: Should I keep a one-liner return for a lower number of lines?

[the remark this answers is not in the document]
Comment (mariacarmina) [reply aquarius/events/util.py:65]: Saw that, `os.getenv` returns None, so one-liner is not a good idea. I will delete the else after the return.

File: aquarius/events/util.py [hunk 5]
```diff
@@ -47,6 +47,24 @@ def get_network_name():
...
+        for network_values in addresses.values()
+        if network_values["chainId"] == chain_id
+    )
+
+
+def get_defined_block(chain_id: int):
+    """Retrieves the block either from envvar, either from address.json file."""
+    if "BFACTORY_BLOCK" in os.environ:
+        return int(os.getenv("BFACTORY_BLOCK"))
+    return get_start_block_by_chain_id(chain_id)
```
Comment (calina-c) [inline aquarius/events/util.py:65]: please add an empty line before `return`.

File: aquarius/events/util.py [hunk 6]
```diff
@@ -47,6 +47,24 @@ def get_network_name():
     return network_name
 
 
+def get_start_block_by_chain_id(chain_id: int) -> str:
+    """Return the contract address with the given name and chain id"""
+    with open(get_address_file(), "r") as address_json:
+        addresses = json.load(address_json)
+    return next(
```
Comment (calina-c) [inline aquarius/events/util.py:54]: please add an empty line before `return`.
Also, do not forget about the doc change related to utf-8.

**Review by alexcos20 — APPROVED** [review]

**Pull request merged**